### PR TITLE
Flag cell as not running when dismissing restart prompt

### LINKED ISSUE #9538: Cancelling the prompt to restarting dead kernels leaves the cell in a bad state
* Run a cell in a notebook
* Kill the kernel
* Attempt to run another cell (or same cell)
* Select the `Cancel` option when prompted to restart the dead kernel
* The cell is still running, when it should not be.

No way to recover, unless we reload VS Code:
* Closing and reopening the notebook still flags the cell as running.
* We can no longer run this cell.
* We can however run other cells (after restarting the kernel).

### PR DESCRIPTION
**Pull request by DonJayamanne**
Fixes #9538

### CODE REVIEW

File: src/platform/errors/errorUtils.node.ts [hunk 1]
```diff
@@ -567,21 +567,26 @@ function isBuiltInModuleOverwritten(
     };
 }
 
-export function displayErrorsInCell(
+export function endCellAndDisplayErrorsInCell(
```
Comment (DonJayamanne) [inline src/platform/errors/errorUtils.node.ts:570]: Renamed to be more explicit about the fact that the cell execution is marked as completed.

File: src/platform/errors/errorUtils.node.ts [hunk 2]
```diff
@@ -567,21 +567,26 @@ function isBuiltInModuleOverwritten(
...
         return;
     }
-    const execution = CellExecutionCreator.getOrCreate(cell, controller);
     // Start execution if not already (Cell execution wrapper will ensure it won't start twice)
-    execution.start(cell.executionSummary?.timing?.endTime);
-    execution.executionOrder = cell.executionSummary?.executionOrder;
-
+    const started = execution.started;
+    if (!started) {
+        execution.start(cell.executionSummary?.timing?.endTime);
```
Comment (DonJayamanne) [inline src/platform/errors/errorUtils.node.ts:587]: Found that we don't need to be restarting if it has already been started

File: src/platform/errors/errorUtils.node.ts [hunk 3]
```diff
@@ -567,21 +567,26 @@ function isBuiltInModuleOverwritten(
...
     cell: NotebookCell,
     controller: NotebookController,
     errorMessage: string,
     isCancelled: boolean
 ) {
+    const execution = CellExecutionCreator.getOrCreate(cell, controller);
     const output = createOutputWithErrorMessageForDisplay(errorMessage);
     if (!output) {
+        if (execution.started) {
+            execution.end(isCancelled ? undefined : false, cell.executionSummary?.timing?.endTime);
```
Comment (DonJayamanne) [inline src/platform/errors/errorUtils.node.ts:580]: This fixes the underlying bug.

**Review by rchiodo — APPROVED** [review]

**Comment by IanMatthewHuff** [comment]
Force resolutions having issues? https://github.com/rogeriochaves/npm-force-resolutions/issues/33? Might just retry that stage.

**Comment by DonJayamanne** [comment]
Will add a test for this, I think we need one.

**Review by amunger — APPROVED** [review]

**Review by IanMatthewHuff — APPROVED** [review]

**Pull request merged**